### DISCUSSION TITLE
Exclude files/folders from live grep search?

### DISCUSSION THREAD

**Discussion by fredrikaverpil** [LazyVim/LazyVim#804] [Q&A] [answered]
Coming from vscode, I miss this the ease of using this box:

![image](https://github.com/LazyVim/LazyVim/assets/994357/1c3292a0-979a-421c-a054-e06ea32c4190)

I'm specially interested in understanding how I can search with live grep but exclude certain files or folders. I came across this blog post: https://miguelcrespo.co/using-telescope-to-find-text-inside-paths

```lua
return {
  "nvim-telescope/telescope.nvim",
  dependencies = {
    { "nvim-telescope/telescope-live-grep-args.nvim" },
  },
  config = function()
    require("telescope").load_extension("live_grep_args")
  end
}
```

```lua
vim.keymap.set('n', '<leader>fs', require("telescope").extensions.live_grep_args.live_grep_args, { noremap = true })
```

Is following those instructions ☝️ my best bet, or does lazyvim have something that makes telescope accept arguments?

Here's another link from the telescope repo advocating for this extension: https://github.com/nvim-telescope/telescope.nvim/issues/2215

**Comment by dpetka2001** [comment]
If you're interested in search/replace capability then I think spectre (which comes with Lazyvim by default) already achieves that by specifying glob patterns in path. If you're interested in live grep for telescope then you'd best go with the plugin they suggested in the topic you linked.

**Reply by fredrikaverpil** [reply]
Ok, great - thank you! 👍

**Reply by cuchoi** [reply]
FYI, the shortcut for this is `<leader>sr` (took me a while to find it)

**Comment by fredrikaverpil** [comment, marked as answer]
For completeness, in the end, I configured live_grep_args to ignore the `.git` folder but search for `--hidden` folders. Then I can use live_grep_args and enter `-g "!somefolder" foo` in the prompt for example, to ignore a certain folder when searching for `foo`.

https://github.com/fredrikaverpil/dotfiles/blob/ee04215d632f7c91287031af41497bae98d63dd8/nvim-lazyvim/lua/plugins/telescope.lua#L43-L54
